src/core/loop_induction_variables/src/InductionVariables.cpp

```diff
@@ -100,16 +102,121 @@ InductionVariableManager::InductionVariableManager(LoopTree *loopNode,
+                foundOnePHI = false;
+                return true;
+              }
+            }
+            return false;
+          });
+
+          if (!foundOnePHI)
+            continue;
+
+          /*
+           * 2. The subloop has only one exit condition, which compares
+           * the subloop's governing IV to a constant.
+           */
+          auto subloop = this->loop->getInnermostLoopThatContains(internalPHI);
+          if (subloop->getLoopExitBasicBlocks().size() != 1)
+            continue;
+
+          if (auto subloopHeaderTerminator =
+                  dyn_cast<BranchInst>(subloop->getHeader()->getTerminator())) {
+            assert(isa<CmpInst>(subloopHeaderTerminator->getCondition())
+                   && "subloop loop exit condition not CmpInst!\n");
+            auto subloopExitCond =
+                cast<CmpInst>(subloopHeaderTerminator->getCondition());
+            auto subloopExitCondL = subloopExitCond->getOperand(0);
+            auto subloopExitCondR = subloopExitCond->getOperand(1);
+
+            const SCEV *subloopIV = nullptr;
+            const SCEV *subloopExitSCEV = nullptr;
+            if (SE.getSCEV(subloopExitCondL)->getSCEVType()
+                    == SCEVTypes::scAddRecExpr
+                && SE.getSCEV(subloopExitCondR)->getSCEVType()
+                       == SCEVTypes::scConstant) {
+              subloopIV = SE.getSCEV(subloopExitCondL);
+              subloopExitSCEV = SE.getSCEV(subloopExitCondR);
+            } else if (SE.getSCEV(subloopExitCondR)->getSCEVType()
+                           == SCEVTypes::scAddRecExpr
+                       && SE.getSCEV(subloopExitCondL)->getSCEVType()
+                              == SCEVTypes::scConstant) {
+              subloopIV = SE.getSCEV(subloopExitCondR);
+              subloopExitSCEV = SE.getSCEV(subloopExitCondL);
+            }
+
+            if (subloopExitSCEV == nullptr || subloopIV == nullptr)
+              continue;
+
+            /*
+             * If all conditions are met, calculate the number of inner loop
+             * iterations (step multiplier) and allocate the IV.
+             */
+            assert(subloopExitSCEV->getSCEVType() == SCEVTypes::scConstant);
+            auto subloopExitConstant =
+                cast<SCEVConstant>(subloopExitSCEV)->getValue()->getSExtValue();
+
+            assert(subloopIV->getSCEVType() == SCEVTypes::scAddRecExpr);
+            auto subloopIVSCEV = cast<SCEVAddRecExpr>(subloopIV);
+
+            if (auto startSCEVConstant =
+                    dyn_cast<SCEVConstant>(subloopIVSCEV->getStart())) {
+              auto subloopStartValue =
+                  startSCEVConstant->getValue()->getSExtValue();
+              if (auto stepSCEVConstant = dyn_cast<SCEVConstant>(
+                      subloopIVSCEV->getStepRecurrence(SE))) {
+                auto subloopStepSize =
+                    stepSCEVConstant->getValue()->getSExtValue();
+                stepMultiplier =
```

Comment:
Are you sure about the correctness of this formula?
What if subloopExitConstant is 2, subloopStartValue is 1, and subloopStepSize is 2? in this case, you execute 1 iteration per invocation of the sub-loop. So shouldn't the stepMultiplier be equal to 1?